multiple .minecraft locations support (+ flatpak default)

Kalitsune opened the pull request:
hi, I've added a multiple minecraft location handling logic (+ it also bring support to flatpak, but not snapcraft as I don't know where the .minecraft folder is located).

It is rather quite modular so it should be pretty easy to add other paths as well.

have a good day,
Kali.

cdbbnnyCode commented:
(Just cleaning up old PR's on this project). This is a good addition. Snap support is most likely not going to be possible (snap data is stored in its own filesystem), but adding checks for different locations is a good idea.

I would say that requiring user input every time the program is launched is not a particularly good idea. It would probably be better to ask once and store the user preference in a file. Additionally, it might be a good idea to add a command line argument to override this preference if necessary.

Kalitsune commented:
Thanks for the response! I've tweaked a few things:
# Updates
- added preferences utils that allow developers to get and write information about a given key
- added description to the already existing `--mcdir` option in the help menu
- the path that should be used will now be asked if no preference is set and no path is provided in the args
- added an `automated` argument that limits blocking interactions (but only implemented it for the path feature)
- if the path provided is different from the one that is saved in the preferences, a prompt ask the user if they want to update the preferences (unless the `--automated` argument is provided)

Let me know if you need anything else to be done ^^
